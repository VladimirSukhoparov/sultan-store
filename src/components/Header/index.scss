$link:#3F4E65;
$white:#FFFFFF;
$orange:#FFC85E;
$lite-grey:#EDEDED;
$black:#111111;

.header {
    padding: 10px 0;

    &__container {
        display: flex;
        flex-direction: column;
        align-content: space-between;
        box-sizing: border-box;
        margin: 0 280px;
    }

    &__link,
    &__contact {
        display: flex;
        align-items: center;
        justify-content: space-between;
        white-space: nowrap;

    }

   

    &__adress,
    &__list,
    &__box {
        display: flex;
        align-items: center;
    }


    &__geo,
    &__mail {
        display: block;
    }

    &__geo {
        width: 13.33px;
        height: 16.67px;
        margin-right: 18px;
    }

    &__mail {
        height: 13.33px;
        width: 16.67px;
        margin-right: 17px;
    }

    &__box {
        margin-right: 42px;

        .up__adress,
        .sub__adress {
            margin: 0;
            padding: 0;
        }

        .up__adress {
            font-weight: 600;
            font-size: 14px;
            line-height: 18.2px;
        }

        .sub__adress {
            font-weight: 300;
            font-size: 12px;
            line-height: 15.6px;
        }


    }


    &__list {
        list-style-type: none;
        padding: 0;
    }

    &__item a {
        text-decoration: none;
        font-weight: 400;
        font-size: 14px;
        line-height: 150%;
        color: $link;
    }

    &__item {
        &:not(:last-child) {
            margin-right: 60px;
        }
    }

    &__contact {
        box-sizing: border-box;
        display: flex;
        align-items: center;
        justify-content: space-between;


    }

    &__logo {
        width: 156px;
        height: 66px;
    }

    &__catalog {
        text-decoration: none;
        color: $white;
        font-weight: 700;
        font-size: 14px;
        line-height: 17px;
        padding: 21px 54px;
        background-color: $orange;
        border-radius: 79px;
        width: 192px;
        height: 59px;
        box-sizing: border-box;

        img {
            width: 15px;
            height: 15px;
            margin-left: 12px;
            margin-bottom: -2px;
        }
    }

    &__input {
        height: 59px;
        border-radius: 36px;
        background-color: $lite-grey;
        vertical-align: middle;
        font-weight: 400;
        font-size: 12px;
        line-height: 150%;
        opacity: 0.6;
        padding: 13px 10px 13px 20px;
        border: none;
        width: 263px;
    }

    &__search {
        position: relative;
    }

    &__searchBtn {
        position: absolute;
        top: 10px;
        left: 214px;
        display: block;
        background-color: $orange;
        width: 39px;
        height: 39px;
        border-radius: 1000px;
        border: none;
        padding-top: 5px;

        img {
            width: 14.53px;
            height: 14.53px;
        }
    }

    &__call {
        display: flex;
        flex-direction: column;
        align-items: end;
        margin-right: 10px;

        p {
            margin: 0;
            padding: 0;
            padding-bottom: 5px;
        }

        .up__call {
            font-weight: 600;
            font-size: 14px;
            line-height: 150%;
            color: $black;
        }

        .sub__call {
            font-weight: 300;
            font-size: 12px;
            line-height: 150%;
            color: $link;
        }

        a {
            font-weight: 400;
            font-size: 10px;
            line-height: 150%;
            color: $link;
            text-decoration: none;
            text-decoration: underline;
        }
    }

    &__caller {
        height: 92px;
        width: 70px;

    }
    #one__line{
        width: 100%;
    }
    #two__line{
        width: 100%;
    }

    &__price {
        text-decoration: none;
        font-weight: 700;
        font-size: 14px;
        line-height: 17px;
        color: $white;
        background-color: $orange;
        border-radius: 79px;
        width: 200px;
        height: 59px;
        box-sizing: border-box;
        padding: 21px 43.5px;

        img {
            margin-left: 14.83px;
            width: 17px;
            height: 17px;
            margin-bottom: -3px;
        }
    }

    &__basket {
        text-decoration: none;
        display: flex;
        align-items: center;
        width: 128px;
        height: 46px;
        position: relative;

        .basket__img {
            width: 46px;
            height: 46px;
            color: $link;
        }

        .basket__num {
            position: absolute;
            top: 0;
            left: 33px;
            box-sizing: border-box;
            width: 28px;
            height: 28px;
            font-weight: 700;
            font-size: 14px;
            line-height: 17px;
            color: $white;
            background: $orange;
            border-radius: 28px;
            border: 3px solid $white;
            margin: 0;
            display: flex;
            align-items: center;
            justify-content: center;

        }

        .basket__box {
            margin-left: 24px;
        }

        .basket__name {
            font-weight: 300;
            font-size: 12px;
            line-height: 150%;
            color: $link;
            margin: 0;
        }

        .basket__price {
            font-weight: 600;
            font-size: 14px;
            line-height: 150%;
            color: $black;
            margin: 0;
        }
    }


    &__tel {
        display: flex;
        align-items: center;
    }

    &__adress {
        width: 50%;
    }

    &__nav {
        width: 50%;
        display: flex;
        flex-direction: row-reverse
    }

    &__content {
        width: 50%;
        display: flex;
        justify-content: space-between;
        align-items: center;

        &:not(:last-child) {
            margin-right: 35px;
        }
    }

    @media (min-width:1360px) and (max-width: 1919px) {
        margin: 0 5%;
        &__container {
            margin: 0;

        }

        &__nav,
        &__adress {
            justify-content: center;
        }

        &__content {
            justify-content: space-around;
        }
    }
    @media (min-width:1024px) and (max-width: 1359px) {
        margin: 0 5%;
       
        &__container {
            margin: 0;

        }
        &__link,&__contact {
        
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;
            
        }
        &__list{
            width: 100%;
            justify-content: space-between;
        }
        &__content{
            width: 100%;
            justify-content: space-between;
        }
    }
    @media (min-width:760px) and (max-width: 1023px) {
        margin: 0 5%;
       
        &__container {
            margin: 0;

        }
        &__link,&__contact {
        
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;

            
        }
        &__list{
            width: 100%;
            justify-content: center;
        }
        &__content{
            width: 100%;
            justify-content: space-between;
        }
    }
    @media (max-width: 759px) {
        margin: 0 5%;
       
        &__container {
            margin: 0;

        }
        &__link,&__contact {
        
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;
            
        }
        &__list{
            width: 100%;
            justify-content: center;
        }
        &__content{
            width: 100%;
            justify-content: space-between;
        }
        &__list{
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;
        }
    }
    @media (max-width: 570px) {
        margin: 0 5%;
       
        &__container {
            margin: 0;

        }
        &__link{
            display: none;
        }
        #one__line{
            display: none;
        }

        &__tel{
            display: none;
        }
        &__price{
            display: none;
        }
        &__contact {
        
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;
            
        }
        &__list{
            width: 100%;
            justify-content: center;
        }
        &__content{
            width: 100%;
            justify-content: space-between;
        }
        &__list{
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            flex-wrap: wrap;
        }
        &__input{
            width: 150px;
        }
        &__searchBtn{
                left: 100px;
        }
    }
}