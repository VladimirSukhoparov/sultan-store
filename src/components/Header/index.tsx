/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState, useEffect, FC } from "react";
import { NavLink } from "react-router-dom";
import { useLocalStorage } from "../../hooks/useLocalStorage";

import "./index.scss";

type headerProps = {
  basket: {
    image: string;
    name: string;
    type: string;
    size: string;
    barcode: number;
    manufacturer: string;
    brand: string;
    description: string;
    price: string;
    typeOfCare: string[];
  }[];
  products: {
    image: string;
    name: string;
    type: string;
    size: string;
    barcode: number;
    manufacturer: string;
    brand: string;
    description: string;
    price: string;
    typeOfCare: string[];
  }[];
  data: {
    image: string;
    name: string;
    type: string;
    size: string;
    barcode: number;
    manufacturer: string;
    brand: string;
    description: string;
    price: string;
    typeOfCare: string[];
  }[];
  setProducts: (product) => void;
};

export const Header:FC<headerProps> = ({ basket, products, data, setProducts }) => {
  const [searchText, setSearchText] = useState("");
  const { readLS, writeLS } = useLocalStorage();

  const localWrite = () => {
    if (!readLS("data")) {
      writeLS("data", data);
    }
  };

  useEffect(() => {
    searchText.length > 0
      ? setProducts(
          products.filter((item) =>
            item.name.toLowerCase().includes(searchText.toLowerCase())
          )
        )
      : setProducts(data);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [searchText]);

  return (
    <div className="header">
      <div className="header__container">
        <div className="header__link">
          <div className="header__box">
            <img
              src={require("../images/geo.png").default}
              alt="geo"
              className="header__geo"
            />
            <div>
              <p className="up__adress">г. Кокчетав, ул. Ж. Ташенова 129Б</p>
              <p className="sub__adress">(Рынок Восточный)</p>
            </div>
          </div>
          <div className="header__box">
            <img
              src={require("../images/mail.png").default}
              alt="geo"
              className="header__mail"
            />
            <div>
              <p className="up__adress">opt.sultan@mail.ru</p>
              <p className="sub__adress">На связи в любое время</p>
            </div>
          </div>

          <ul className="header__list">
            <li className="header__item">
              <a>О компании</a>
            </li>
            <li className="header__item">
              <a>Доставка и оплата</a>
            </li>
            <li className="header__item">
              <a>Возврат</a>
            </li>
            <li className="header__item">
              <a>Контакты</a>
            </li>
          </ul>
        </div>

        <hr id="one__line" />
        <div className="header__contact">
          {/* <div className="header__content"> */}
          <NavLink to="/">
            <img
              src={require("../images/logo.png").default}
              alt="logo"
              className="header__logo"
            />
          </NavLink>
          <NavLink
            to="catalog"
            className="header__catalog"
            onClick={localWrite}
          >
            Каталог
            <img src={require("../images/catalog.png").default} alt="catalog" />
          </NavLink>
          <div className="header__search">
            <input
              type="search"
              className="header__input"
              placeholder="Поиск..."
              onChange={(event) => setSearchText(event.target.value)}
            />
            <button className="header__searchBtn">
              <img src={require("../images/search.png").default} alt="search" />
            </button>
          </div>

          {/*  </div>
          <div className="header__content"> */}
          <div className="header__tel">
            <div className="header__call">
              <p className="up__call">+7 (777) 490-00-91</p>
              <p className="sub__call">время работы: 9:00-20:00</p>
              <a>Заказать звонок</a>
            </div>
            <img
              src={require("../images/call.png").default}
              alt="caller"
              className="header__caller"
            />
          </div>
          <a className="header__price">
            Прайс-лист
            <img
              src={require("../images/download.png").default}
              alt="download"
            />
          </a>
          <NavLink to="basket" className="header__basket">
            <img
              src={require("../images/basket.png").default}
              alt="basket"
              className="basket__img"
            />
            <p className="basket__num">{basket.length}</p>
            <div className="basket__box">
              <p className="basket__name">Корзина</p>
              <p className="basket__price">
                {Math.floor(
                  basket.reduce((acc, cur) => acc + parseFloat(cur.price), 0) *
                    100
                ) / 100}{" "}
                ₸
              </p>
            </div>
          </NavLink>
          {/* </div> */}
        </div>
        <hr id="two__line" />
      </div>
    </div>
  );
};
