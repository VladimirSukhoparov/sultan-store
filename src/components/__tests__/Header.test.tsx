import { screen, render, cleanup } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Header } from "../Header";

afterEach(() => {
  cleanup();
});

test("should render header component", () => {
  render(
    <BrowserRouter>
      <Header
        basket={[]}
        products={[]}
        data={[]}
        setProducts={function (product: any): void {}}
      />
    </BrowserRouter>
  );
  const headerElement = screen.getByTestId("header");
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).toHaveTextContent("Корзина");
});

test("matches snapshot", () => {
  const header = renderer
    .create(
      <BrowserRouter>
        <Header
          basket={[]}
          products={[]}
          data={[]}
          setProducts={function (product: any): void {}}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(header).toMatchSnapshot();
});
