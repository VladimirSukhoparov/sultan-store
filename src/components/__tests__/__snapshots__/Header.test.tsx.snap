// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="header"
  data-testid="header"
>
  <div
    className="header__container"
  >
    <div
      className="header__link"
    >
      <div
        className="header__box"
      >
        <img
          alt="geo"
          className="header__geo"
          src="geo.png"
        />
        <div>
          <p
            className="up__adress"
          >
            г. Кокчетав, ул. Ж. Ташенова 129Б
          </p>
          <p
            className="sub__adress"
          >
            (Рынок Восточный)
          </p>
        </div>
      </div>
      <div
        className="header__box"
      >
        <img
          alt="geo"
          className="header__mail"
          src="mail.png"
        />
        <div>
          <p
            className="up__adress"
          >
            opt.sultan@mail.ru
          </p>
          <p
            className="sub__adress"
          >
            На связи в любое время
          </p>
        </div>
      </div>
      <ul
        className="header__list"
      >
        <li
          className="header__item"
        >
          <a>
            О компании
          </a>
        </li>
        <li
          className="header__item"
        >
          <a>
            Доставка и оплата
          </a>
        </li>
        <li
          className="header__item"
        >
          <a>
            Возврат
          </a>
        </li>
        <li
          className="header__item"
        >
          <a>
            Контакты
          </a>
        </li>
      </ul>
    </div>
    <hr
      id="one__line"
    />
    <div
      className="header__contact"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          className="header__logo"
          src="logo.png"
        />
      </a>
      <a
        className="header__catalog"
        href="/catalog"
        onClick={[Function]}
      >
        Каталог
        <img
          alt="catalog"
          src="catalog.png"
        />
      </a>
      <div
        className="header__search"
      >
        <input
          className="header__input"
          onChange={[Function]}
          placeholder="Поиск..."
          type="search"
        />
        <button
          className="header__searchBtn"
        >
          <img
            alt="search"
            src="search.png"
          />
        </button>
      </div>
      <div
        className="header__tel"
      >
        <div
          className="header__call"
        >
          <p
            className="up__call"
          >
            +7 (777) 490-00-91
          </p>
          <p
            className="sub__call"
          >
            время работы: 9:00-20:00
          </p>
          <a>
            Заказать звонок
          </a>
        </div>
        <img
          alt="caller"
          className="header__caller"
          src="call.png"
        />
      </div>
      <a
        className="header__price"
      >
        Прайс-лист
        <img
          alt="download"
          src="download.png"
        />
      </a>
      <a
        className="header__basket"
        href="/basket"
        onClick={[Function]}
      >
        <img
          alt="basket"
          className="basket__img"
          src="basket.png"
        />
        <p
          className="basket__num"
        >
          0
        </p>
        <div
          className="basket__box"
        >
          <p
            className="basket__name"
          >
            Корзина
          </p>
          <p
            className="basket__price"
          >
            0
             
            ₸
          </p>
        </div>
      </a>
    </div>
    <hr
      id="two__line"
    />
  </div>
</div>
`;
